src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -454,6 +462,26 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     this.alreadyHandlingDeviceChange = false;
   }
 
+  private async handleDeviceStreamEnded(kind: string, deviceId: string): Promise<void> {
+    if (
+      kind === 'video' &&
+      this.boundAudioVideoController &&
+      this.boundAudioVideoController.videoTileController.hasStartedLocalVideoTile()
+    ) {
+      this.boundAudioVideoController.videoTileController.stopLocalVideoTile();
+    }
+    await this.chooseInputDevice(kind, null, false);
```

Comment:
`chooseInputDevice` is a long method. (Hope we can break it down soon.) I wonder if we want to put it into `try`/`catch` block.

Reply:
I would prefer to leave it as is as it is how it was before and we usually use it like that in other places.

Reply:
Before this PR, we are calling it asynchronously without `await`, so the next line is guaranteed to be called. In this PR, the observer lines won't be reached.

```
this.chooseInputDevice(kind, null, false);
if (
  kind === 'video' &&
  this.boundAudioVideoController &&
  this.boundAudioVideoController.videoTileController.hasStartedLocalVideoTile()
) {
  this.boundAudioVideoController.videoTileController.stopLocalVideoTile();
}
```

Reply:
Trung and I decided to skip the `try/catch` block here as `chooseInputDevice` is catching thrown errors from browser API calls.